### PR TITLE
Add Portal to the Date component

### PR DESCRIPTION
**Pull request by paulsturgess**
# Description
The Date Picker now uses Portal to handle the layout.

Note that in order to write the specs I needed to add a ref to the Portal component itself. This is so that the content inside the Portal can be found using Enzyme as the content has been pulled out of the dom tree of the parent component.

# TODO
- [x] Release notes
- [x] Repoint this branch to master

# Related Issues / Pull Requests
PR that introduces Portal: https://github.com/Sage/carbon/pull/1542
CARBON-967

# Testing Instructions
No visible changes, ensure the date picker behaves as it currently does.

### CODE REVIEW

File: src/components/date/date.js [hunk 1]
```diff
@@ -500,14 +516,66 @@ class Date extends React.Component {
...
    * Returns the DayPicker component
    *
    * @method renderDatePicker
    * @return {Object} JSX
    */
   renderDatePicker() {
     if (!this.state.open) { return null; }
-    return <DayPicker { ...this.datePickerProps } />;
+    return (
+      <Portal open>
```
Comment (SeanArmstrong) [inline src/components/date/date.js]: Do we just want the portal to deal with the open state. Rather than doing a check in the datepicker?

Comment (paulsturgess) [reply src/components/date/date.js]: @SeanArmstrong done

**Comment by paulsturgess** [comment]
Note this PR is currently pointing at https://github.com/Sage/carbon/pull/1542 as I branched from there due it implementing the `Portal` component.

**Comment by ianrobsonsage** [comment]
Pointed GAC at the portal_date_picker branch of Carbon.

Tested the Date Pickers in customer and Supplier screens, no regressions, behaviour as expected. Tested cross browser.

Qa Passed

**Pull request merged**